monitoring/monitoring_internal_test.go

```diff
@@ -181,528 +192,543 @@ func TestPeriodicReport(t *testing.T) {
+		PollPeriod: config.Duration{Duration: duration},
+	}
+
+	testData := []testAlertData{
+		{
+			trafficMonitoring: testTrafficMonitoring{inputTraffic: 150, outputTraffic: 150},
+			monitoringData: cloudprotocol.MonitoringData{
+				Global: cloudprotocol.GlobalMonitoringData{
+					RAM:        1100,
+					CPU:        35,
+					UsedDisk:   2300,
+					InTraffic:  150,
+					OutTraffic: 150,
 				},
-				InTraffic: &config.AlertRule{
-					MinTimeout:   config.Duration{},
-					MinThreshold: 0,
-					MaxThreshold: 0,
+			},
+			quotaData: testQuotaData{
+				cpu:  35,
+				ram:  1100,
+				disk: 2300,
+			},
+			alertExpectedCount: 0,
+		},
+		{
+			trafficMonitoring: testTrafficMonitoring{inputTraffic: 150, outputTraffic: 250},
+			monitoringData: cloudprotocol.MonitoringData{
+				Global: cloudprotocol.GlobalMonitoringData{
+					RAM:        1100,
+					CPU:        45,
+					UsedDisk:   2300,
+					InTraffic:  150,
+					OutTraffic: 250,
 				},
-				OutTraffic: &config.AlertRule{
-					MinTimeout:   config.Duration{},
-					MinThreshold: 0,
-					MaxThreshold: 0,
+			},
+			quotaData: testQuotaData{
+				cpu:  45,
+				ram:  1100,
+				disk: 2300,
+			},
+			alertExpectedCount: 2,
+			alert: []cloudprotocol.AlertItem{
+				prepareSystemAlertItem("cpu", time.Now(), 45),
+				prepareSystemAlertItem("outTraffic", time.Now(), 250),
+			},
+		},
+		{
+			trafficMonitoring: testTrafficMonitoring{inputTraffic: 350, outputTraffic: 250},
+			monitoringData: cloudprotocol.MonitoringData{
+				Global: cloudprotocol.GlobalMonitoringData{
+					RAM:        2100,
+					CPU:        45,
+					UsedDisk:   4300,
+					InTraffic:  350,
+					OutTraffic: 250,
 				},
 			},
+			quotaData: testQuotaData{
+				cpu:  45,
+				ram:  2100,
+				disk: 4300,
+			},
+			alertExpectedCount: 5,
+			alert: []cloudprotocol.AlertItem{
+				prepareSystemAlertItem("cpu", time.Now(), 45),
+				prepareSystemAlertItem("ram", time.Now(), 2100),
+				prepareSystemAlertItem("disk", time.Now(), 4300),
+				prepareSystemAlertItem("inTraffic", time.Now(), 350),
+				prepareSystemAlertItem("outTraffic", time.Now(), 250),
+			},
 		},
-		sender,
-		networkManager)
-	if err != nil {
-		t.Fatalf("Can't create monitoring instance: %s", err)
 	}
-	defer monitor.Close()
 
-	for {
+	for _, item := range testData {
+		trafficMonitoring = item.trafficMonitoring
+		systemQuotaData = item.quotaData
+
+		monitor, err := New(
+			&config.Config{
+				WorkingDir: ".",
+				Monitoring: monitoring,
+			}, sender, &trafficMonitoring)
+		if err != nil {
+			t.Fatalf("Can't create monitoring instance: %s", err)
+		}
+
 		select {
-		case <-monitor.GetMonitoringDataChannel():
-			for resource, numAlerts := range alertMap {
-				if numAlerts != 1 {
-					t.Errorf("Wrong number of %s alerts: %d", resource, numAlerts)
-				}
+		case monitoringData := <-monitor.GetMonitoringDataChannel():
+			if len(alertMap) != item.alertExpectedCount {
+				t.Error("Alerts should not be generated")
```

Comment:
Should be fatal here. Otherwise following code may crash in case alertMap size more than currentAlert.
Fix error message.

Reply:
Done